sklearn/model_selection/tests/test_classification_threshold.py

```diff
@@ -682,3 +684,49 @@ def test_fixed_threshold_classifier_metadata_routing():
     classifier_default_threshold = FixedThresholdClassifier(estimator=clone(classifier))
     classifier_default_threshold.fit(X, y, sample_weight=sample_weight)
     assert_allclose(classifier_default_threshold.estimator_.coef_, classifier.coef_)
+
+
+@pytest.mark.parametrize(
+    "scoring_name", ["roc_auc", "average_precision", "neg_log_loss"]
+)
+def test_error_on_unthresholded_classification_metrics(scoring_name):
+    """Check that we raise an error if optimizing untresholded metrics."""
+    X, y = make_classification(random_state=0)
+    estimator = LogisticRegression()
+    if scoring_name in ("roc_auc", "average_precision"):
+        expected_method_names = "('decision_function', 'predict_proba')"
+    else:
+        expected_method_names = "'predict_proba'"
```

Comment:
Could you add `expected_method_name` to `@pytest.mark.parametrize`?

Reply:
Done.